worlds/yoshisisland/level_logic.py

```diff
@@ -0,0 +1,483 @@
+        if self.game_logic != "Easy":
+            return True
+        else:
+            return (self.default_vis(state) or state.has('Secret Lens', self.player) or self.combat_item(state))
+
+    def bowserdoor_1(self, state: CollectionState) -> bool:
+        if self.game_logic == "Easy":
+            return state.has_all({'Egg Plant', '! Switch'}, self.player) and state.has('Egg Capacity Upgrade', self.player, 2)
+        elif self.game_logic == "Normal":
+            return state.has_all({'Egg Plant'}, self.player) and state.has('Egg Capacity Upgrade', self.player, 1)
+        else:
+            return state.has_all({'Egg Plant'}, self.player)
+
+    def bowserdoor_2(self, state: CollectionState) -> bool:
+        if self.game_logic == "Easy":
+            return ((state.has('Egg Capacity Upgrade', self.player, 3) and state.has('Egg Plant', self.player)) or self.combat_item(state)) and state.has_all({'Key'}, self.player)
+        elif self.game_logic == "Normal":
+            return ((state.has('Egg Capacity Upgrade', self.player, 2) and state.has('Egg Plant', self.player)) or self.combat_item(state)) and state.has_all({'Key'}, self.player)
+        else:
+            return ((state.has('Egg Capacity Upgrade', self.player, 1) and state.has('Egg Plant', self.player)) or self.combat_item(state)) and state.has_all({'Key'}, self.player)
+
+    def bowserdoor_3(self, state: CollectionState) -> bool:
+        if self.game_logic == "Easy":
+            return True
+        elif self.game_logic == "Normal":
+            return True
+        else:
+            return True
+
+    def bowserdoor_4(self, state: CollectionState) -> bool:
+        if self.game_logic == "Easy":
+            return True
+        elif self.game_logic == "Normal":
+            return True
+        else:
+            return True
+
+    def _68Route(self, state: CollectionState) -> bool:
+        if self.bowser_door == 0:
+            return True
+        elif self.bowser_door == 1:
+            return self.bowserdoor_1(state)
+        elif self.bowser_door == 2:
+            return self.bowserdoor_2(state)
+        elif self.bowser_door == 3:
+            return True
+        elif self.bowser_door == 4:
+            return True
+        elif self.bowser_door == 5:
+            return self.bowserdoor_1(state) and self.bowserdoor_2(state) and self.bowserdoor_3(state)
+
+    def _68CollectibleRoute(self, state: CollectionState) -> bool:
+        if self.bowser_door == 0:
+            return True
+        elif self.bowser_door == 1:
+            return self.bowserdoor_1(state)
+        elif self.bowser_door == 2:
+            return self.bowserdoor_2(state)
+        elif self.bowser_door == 3:
+            return True
+        elif self.bowser_door == 4:
+            return True
+        elif self.bowser_door == 5:
+            return self.bowserdoor_1(state)
+
+
+##############################################################################

```

Comment:
What are these random lines of `#`?

Reply:
They are for ease of readability; separating the actual specific logic functions from the level-based logic ones, just a visual thing. Was more necessary when every level had its own function before being moved to rules.

Reply:
While code organization is a very subjective concept, a good rule of thumb is that once you reach a point where you try to insert some strange, non-standard stuff to make it more readable, is the point where you should instead reorganise it.

For example, here, if you have methods for level based logic, and methods for the rest of logic, why not have a file/class for level logic and a different file/class for other logic stuff? This way, the code is organized the way **you** want it, while not having weird lines of nothing polluting it